### PR TITLE
Allow use of "finalized" tag in rpc requests

### PR DESCRIPTION
**Pull request by piersy**
This PR allows the use of the "finalized" tag to retrieve blocks or
headers via the RPC. Using the "finalized" tag is equivalent to using
the "latest" tag because all blocks are final in celo.

This is to supprt integration with chainlink since their system depends
on retrieving finalized blocks.

### Tested

There is a new e2e test that checks the "finalized" tag retrieves the same header as the "latest" tag.

### CODE REVIEW

File: rpc/types_test.go [hunk 1]
```diff
@@ -87,18 +88,20 @@ func TestBlockNumberOrHash_UnmarshalJSON(t *testing.T) {
...
-		19: {`{"blockHash":"0x0000000000000000000000000000000000000000000000000000000000000000","requireCanonical":false}`, false, BlockNumberOrHashWithHash(common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000000"), false)},
-		20: {`{"blockHash":"0x0000000000000000000000000000000000000000000000000000000000000000","requireCanonical":true}`, false, BlockNumberOrHashWithHash(common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000000"), true)},
-		21: {`{"blockNumber":"0x1"}`, false, BlockNumberOrHashWithNumber(1)},
-		22: {`{"blockNumber":"pending"}`, false, BlockNumberOrHashWithNumber(PendingBlockNumber)},
-		23: {`{"blockNumber":"latest"}`, false, BlockNumberOrHashWithNumber(LatestBlockNumber)},
-		24: {`{"blockNumber":"earliest"}`, false, BlockNumberOrHashWithNumber(EarliestBlockNumber)},
-		25: {`{"blockNumber":"0x1", "blockHash":"0x0000000000000000000000000000000000000000000000000000000000000000"}`, true, BlockNumberOrHash{}},
+		14: {`"finalized"`, false, BlockNumberOrHashWithNumber(EarliestBlockNumber)},
```
Comment (carterqw2) [inline rpc/types_test.go:91]: nit: why `EasliestBlockNumber` here?

Comment (piersy) [reply rpc/types_test.go:91]: Not a nit, that's a copy paste error. Thanks!

File: rpc/types.go [hunk 2]
```diff
@@ -106,6 +108,9 @@ func (bn *BlockNumber) UnmarshalJSON(data []byte) error {
 	case "pending":
 		*bn = PendingBlockNumber
 		return nil
+	case "finalized":
```
Comment (carterqw2) [inline rpc/types.go:111]: Looks like you forgot to add it to `func (bnh *BlockNumberOrHash) UnmarshalJSON(data []byte) error {`

Comment (piersy) [reply rpc/types.go:111]: Yes, its added now. Thanks!

**Review by carterqw2 — APPROVED** [review]

**Pull request merged**